src/http.rs

```diff
@@ -170,11 +170,16 @@ impl Client {
         } else {
             Cookie::default()
         };
+        let mut requests = vec![pipeline::StreamRequest::Execute(
+            pipeline::StreamExecuteReq { stmt },
+        )];
+        if tx_id == 0 {
+            // automatically close one-shot requests
+            requests.push(pipeline::StreamRequest::Close);
```

Comment:
nitpick: let's rewrite it to
```
let mut requests = if tx_id == 0 {
    vec![...]
} else {
    vec![...]
}
```
, otherwise the compiler could allocate on push for oneshot requests, and they should be on the fast path. Better be on the safe side here